### ISSUE TITLE
[FEATURE]: Clear Instruction for Checkpoint

### ISSUE THREAD

**Issue by binmakeswell** [hpcaitech/ColossalAI#2638]
### Describe the feature

For users,  [tutorial](https://www.colossalai.org/docs/basics/model_checkpoint) and [code](https://github.com/hpcaitech/ColossalAI/blob/main/colossalai/utils/checkpointing.py#L154) of checkpointing are not clear enough how to use checkpoint. The code comments are misleading.

Our checkpointing mainly focus on handing model. But users also needs optimizer and lr scheduler states, which is required load_state_dict.

**Comment by liuslnlp** [comment]
Yes, users hope to have a demo to show how to save (and load) the states of the model, optimizer, and lr scheduler in hybrid parallel scenarios, so as to continue training from the last checkpoint after a downtime. 

I suggest add an example in `examples/language/gpt`.

**Comment by binmakeswell** [comment]
用户反馈：目前ColossalAI在如何保存（和加载）混合并行模式下（zero3+tensor+流水线）的优化器和lr调度器参数还没有一个完整的example，这个特性对宕机后从最后一个checkpoint重启训练很重要，可不可以麻烦工作人员补充一下demo。

**Comment by liuslnlp** [comment]
I suggest creating two unified functions to save and load the above parameters. The storage format can follow the example of deepspeed: each worker only stores and loads its corresponding model parameters (zero3) and optimizer status (Figure 1), and provides a conversion script to facilitate users to aggregate model parameters after training (Figure 2).
![image](https://user-images.githubusercontent.com/17002231/219011893-882d10d0-c9bb-4b4d-80a4-2ebf73140553.png)
![image](https://user-images.githubusercontent.com/17002231/219011958-0a0e92c4-6273-4594-be5f-2d8bcc289a6a.png)

**Comment by hijkzzz** [comment]
Could you share this `zero_to_fp32` script ?

> I suggest creating two unified functions to save and load the above parameters. The storage format can follow the example of deepspeed: each worker only stores and loads its corresponding model parameters (zero3) and optimizer status (Figure 1), and provides a conversion script to facilitate users to aggregate model parameters after training (Figure 2). ![image](https://user-images.githubusercontent.com/17002231/219011893-882d10d0-c9bb-4b4d-80a4-2ebf73140553.png) ![image](https://user-images.githubusercontent.com/17002231/219011958-0a0e92c4-6273-4594-be5f-2d8bcc289a6a.png)

**Comment by FrankLeeeee** [comment]
> 

we are designing the new checkpoint io module to support checkpoint saving/loading from various formats, such as single model weights, huggingface style sharded weights and megatron-style sharded tensor weights, etc. Stay tuned.

**Comment by hijkzzz** [comment]
Is there an ETA here?

> > 
> 
> we are designing the new checkpoint io module to support checkpoint saving/loading from various formats, such as single model weights, huggingface style sharded weights and megatron-style sharded tensor weights, etc. Stay tuned.

**Comment by binmakeswell** [comment]
We have completed most of the related development and are doing the final polishing and refinement. Thanks.
